app/controllers/work_packages/auto_completes_controller.rb

```diff
@@ -30,43 +30,65 @@
 require 'rack/utils'
 
 class WorkPackages::AutoCompletesController < ApplicationController
-  before_filter :find_project
-
   def index
-    @work_packages = []
-    q = params[:q].to_s
+    scope = determine_scope
+    if scope.nil?
+      render_404
+      return
+    end
 
-    if q.present?
-      query = (params[:scope] == 'all' && Setting.cross_project_work_package_relations?) ? WorkPackage : @project.work_packages
+    query_term = params[:q].to_s
+    @work_packages = []
+    # query for exact ID matches first, to make an exact match the first result of autocompletion
+    if query_term =~ /\A\d+\z/
+      @work_packages |= scope.visible.find_all_by_id(query_term.to_i)
+    end
 
-      @work_packages |= query.visible.find_all_by_id(q.to_i) if q =~ /\A\d+\z/
+    sql_query = ["LOWER(#{WorkPackage.table_name}.subject) LIKE :q OR
+                  CAST(#{WorkPackage.table_name}.id AS CHAR(13)) LIKE :q",
+                 { q: "%#{query_term.downcase}%" }]
 
-      @work_packages |= query.visible.find(:all,
-                                           limit: 10,
-                                           order: "#{WorkPackage.table_name}.id ASC",
-                                           conditions: ["LOWER(#{WorkPackage.table_name}.subject) LIKE :q OR CAST(#{WorkPackage.table_name}.id AS CHAR(13)) LIKE :q", { q: "%#{q.downcase}%" }])
-    end
+    @work_packages |= scope.visible
+                           .where(sql_query)
+                           .order(id: :asc)
```

Comment:
<pre>

ActiveRecord::StatementInvalid (PG::SyntaxError: ERROR:  non-integer constant in ORDER BY
LINE 2: ...AST(work_packages.id AS CHAR(13)) LIKE '%45%') ORDER BY '---
                                                                   ^
: SELECT  "work_packages"."id" AS t0_r0, "work_packages"."type_id" AS t0_r1, "work_packages"."project_id" AS t0_r2, "work_packages"."subject" AS t0_r3, "work_packages"."description" AS t0_r4, "work_packages"."due_date" AS t0_r5, "work_packages"."category_id" AS t0_r6, "work_packages"."status_id" AS t0_r7, "work_packages"."assigned_to_id" AS t0_r8, "work_packages"."priority_id" AS t0_r9, "work_packages"."fixed_version_id" AS t0_r10, "work_packages"."author_id" AS t0_r11, "work_packages"."lock_version" AS t0_r12, "work_packages"."done_ratio" AS t0_r13, "work_packages"."estimated_hours" AS t0_r14, "work_packages"."created_at" AS t0_r15, "work_packages"."updated_at" AS t0_r16, "work_packages"."start_date" AS t0_r17, "work_packages"."parent_id" AS t0_r18, "work_packages"."responsible_id" AS t0_r19, "work_packages"."root_id" AS t0_r20, "work_packages"."lft" AS t0_r21, "work_packages"."rgt" AS t0_r22, "projects"."id" AS t1_r0, "projects"."name" AS t1_r1, "projects"."description" AS t1_r2, "projects"."homepage" AS t1_r3, "projects"."is_public" AS t1_r4, "projects"."parent_id" AS t1_r5, "projects"."created_on" AS t1_r6, "projects"."updated_on" AS t1_r7, "projects"."identifier" AS t1_r8, "projects"."status" AS t1_r9, "projects"."lft" AS t1_r10, "projects"."rgt" AS t1_r11, "projects"."summary" AS t1_r12, "projects"."project_type_id" AS t1_r13, "projects"."responsible_id" AS t1_r14, "projects"."work_packages_responsible_id" AS t1_r15 FROM "work_packages" LEFT OUTER JOIN "projects" ON "projects"."id" = "work_packages"."project_id" WHERE (projects.status=1 AND projects.id IN (SELECT em.project_id FROM enabled_modules em WHERE em.name='work_package_tracking')) AND (LOWER(work_packages.subject) LIKE '%45%' OR
                    CAST(work_packages.id AS CHAR(13)) LIKE '%45%') ORDER BY '---
:id: :asc
' LIMIT 10):
  app/controllers/work_packages/auto_completes_controller.rb:58:in `|'
  app/controllers/work_packages/auto_completes_controller.rb:58:in `index'
  app/middleware/params_parser_with_exclusion.rb:40:in `call'

</pre>